### DISCUSSION TITLE
504 Gateway Time-out if i use api

### DISCUSSION THREAD

**Discussion by Sivolen** [netbox-community/netbox#10927] [Q&A]
Hi!

I have a problem, when I request via api with the ?limit=0 parameter, I get a 504 error, how can I fix this ?

**Comment by Sivolen** [comment]
I changed nginx settings and it worked

**Reply by pettcomputers** [reply]
@Sivolen What was your fix? What changes did you make to the nginx config?
I'm running into the same error with NetBox returning 504 Gateway Time-out

**Reply by jeremystretch** [reply]
This can be alleviated somewhat by allowing a longer timeout in the [gunicorn configuration](https://github.com/netbox-community/netbox/blob/92ec06c6948f0b6f3c6192f4a1be57a505930ce4/contrib/gunicorn.py#L12) (or whichever WSGI server you're using).

**Reply by Sivolen** [reply]
> @Sivolen What was your fix? What changes did you make to the nginx config? I'm running into the same error with NetBox returning 504 Gateway Time-out

I added in my nginx config
    proxy_connect_timeout 600;
    proxy_send_timeout 600;
    proxy_read_timeout 600;
    send_timeout 600;

**Comment by LHBL2003** [comment]
We also encountered this problem when exporting 4,765 devices to CSV, which resulted in a 5MB CSV file. The 600-second timeout helped us solve the problem. Thank you very much.
If anyone needs instructions on exactly how to proceed:
Here are the steps to follow if NetBox was installed on an Ubuntu system according to the official documentation using options A.

# Error Message

504 Gateway Time-out nginx/1.18.0 (Ubuntu)

<img width="507" height="132" alt="image" src="https://github.com/user-attachments/assets/c0c7e911-5134-4e98-a8ef-39504797b1ba" />


# Instructions
- Open the Nginx configuration as an administrator:
/etc/nginx/sites-available/netbox

Insert this block::
```
	proxy_connect_timeout 600;
	proxy_send_timeout 600;
	proxy_read_timeout 600;
	send_timeout 600;
```

Example:
```
server {
    listen [::]:443 ssl ipv6only=off;

    # CHANGE THIS TO YOUR SERVER'S NAME
    server_name NETBOX.MyDomain.local;

    ssl_certificate /etc/ssl/certs/netbox.crt;
    ssl_certificate_key /etc/ssl/private/netbox.key;

    client_max_body_size 25m;

    location /static/ {
        alias /opt/netbox/netbox/static/;
    }

    location / {
        proxy_pass http://127.0.0.1:8001;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
        
        # Custom: To avoid error 594 Gateway Timeout when loading all devices. 
        # See also: https://github.com/netbox-community/netbox/discussions/10927
        proxy_connect_timeout 600;
        proxy_send_timeout 600;
        proxy_read_timeout 600;
        send_timeout 600;
    }
}

#server {
#    # Redirect HTTP traffic to HTTPS
#    listen [::]:80 ipv6only=off;
#    server_name _;
#    return 301 https://$host$request_uri;
#}
```

Restart Nginx:
```
sudo nginx -t
sudo systemctl reload nginx
```
